isobus/include/isobus/isobus/isobus_time_date_interface.hpp

```diff
@@ -0,0 +1,137 @@
+//================================================================================================
+/// @file isobus_time_date_interface.hpp
+///
+/// @brief Defines an interface for accessing or sending time and date information using
+/// the Time/Date (TD) PGN. Can be useful for interacting with an ISOBUS file server,
+/// or just for keeping track of time and date information as provided by some authoritative
+/// control function on the bus. Control functions which provide the message this interface
+/// manages are expected to have a real-time clock (RTC) or GPS time source.
+/// @author Adrian Del Grosso
+///
+/// @copyright 2024 The Open-Agriculture Developers
+//================================================================================================
+#ifndef ISOBUS_TIME_DATE_INTERFACE_HPP
+#define ISOBUS_TIME_DATE_INTERFACE_HPP
+
+#include "isobus/isobus/can_callbacks.hpp"
+#include "isobus/isobus/can_message.hpp"
+#include "isobus/utility/event_dispatcher.hpp"
+
+namespace isobus
+{
+	class InternalControlFunction;
+	class ControlFunction;
```

Comment:
thought: is it better to forward-declare than to just include it?

Reply:
I mean, not besides it's less for the compiler to pull in when it builds the file. Not really a functional difference in this case, I could just include InternalControlFunction's header if ya like